packages/office-ui-fabric-react/src/components/Slider/Slider.base.tsx

```diff
@@ -0,0 +1,329 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  KeyCodes,
+  getId,
+  getRTL,
+  getRTLSafeKeyCode,
+  createRef,
+  customizable,
+  classNamesFunction,
+  IClassNames
+} from '../../Utilities';
+import { ISliderProps, ISlider, ISliderStyleProps, ISliderStyles } from './Slider.types';
+import { Label } from '../../Label';
+import * as stylesImport from './Slider.scss';
+const styles: any = stylesImport;
+
+export interface ISliderState {
+  value?: number;
+  renderedValue?: number;
+}
+
+export enum ValuePosition {
+  Previous = 0,
+  Next = 1
+}
+
+const getClassNames = classNamesFunction<ISliderStyleProps, ISliderStyles>();
+
+/**
+* Slider with no default styles.
+* [Use the `getStyles` API to add your own styles.](https://github.com/OfficeDev/office-ui-fabric-react/wiki/Styling)
+*/
+@customizable('Slider', ['theme'])
+export class SliderBase extends BaseComponent<ISliderProps, ISliderState> implements ISlider {
+  public static defaultProps: Partial<ISliderProps> = {
+    step: 1,
+    min: 0,
+    max: 10,
+    showValue: true,
+    disabled: false,
+    vertical: false,
+    buttonProps: {}
+  };
+
+  private _classNames: IClassNames<ISliderStyles>;
+  private _sliderLine = createRef<HTMLDivElement>();
+  private _thumb = createRef<HTMLSpanElement>();
+  private _id: string;
+
+  constructor(props: ISliderProps) {
+    super(props);
+
+    this._warnMutuallyExclusive({
+      'value': 'defaultValue'
+    });
+
+    this._id = getId('Slider');
+
+    const value = props.value || props.defaultValue || props.min;
+
+    this.state = {
+      value: value,
+      renderedValue: value
+    };
+  }
+
+  /**
+   * Invoked when a component is receiving new props. This method is not called for the initial render.
+   */
+  public componentWillReceiveProps(newProps: ISliderProps): void {
+
+    if (newProps.value !== undefined) {
+      const value = Math.max(newProps.min as number, Math.min(newProps.max as number, newProps.value));
+
+      this.setState({
+        value: value,
+        renderedValue: value
+      });
+    }
+  }
+
+  public render(): JSX.Element {
```

Comment:
While here updated `React.ReactElement<{}>` to `JSX.Element`. Will split out if problematic re: backwards compatibility